src/components/common/API.jsx

```diff
@@ -292,13 +292,26 @@ class API {
     }
 
     getActiveLoomMetadataEmbeddings() {
-        return this.loomFiles[this.activeLooms[0]].cellMetaData.embeddings;
+        return new Promise((resolve) => {
+            if (
+                typeof this.loomFiles !== 'undefined' &&
+                this.activeLooms[0] in this.loomFiles
+            ) {
+                resolve(
+                    this.loomFiles[this.activeLooms[0]].cellMetaData.embeddings
+                );
+            } else {
+                setTimeout(this.getActiveLoomMetadataEmbeddings, 250);
```

Comment:
My only (minor - for now) concern is that this will never end.

Reply:
See comment below, this was flawed anyway and should be addressed in its own PR